### PR TITLE
feat(ec2): add `versionDescription` property for `LaunchTemplate`

### PR DESCRIPTION
**Pull request by yuppe-kyupeen**
### Reason for this change
The change introduces the `versionDescription` property to the `LaunchTemplate`

### Description of changes
- Add the `versionDescription` property for `LaunchTemplateProps`, which was missing in the L2 construct.
- Add validation for character limit 

### Description of how you validated changes
I Added a unit test for launch template and added the `versionDescription` property in the integration tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts [hunk 1]
```diff
@@ -221,6 +221,15 @@ export interface LaunchTemplateProps {
...
   readonly launchTemplateName?: string;
 
+  /**
+   * A description for the first version of the launch template.
+   * The version descrioption must be maximum 255 characters long, including hyphens (-),
+   * underscores (_), spaces, and tabs.
+   *
+   * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-launchtemplate.html#cfn-ec2-launchtemplate-versiondescription
+   */
+  readonly versionDescription?: string;
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts:231]: The CodeBuild CI seems to be failing. Optional properties must have `@default` documentation. Could you add the label and description?

```
aws-cdk-lib: error: [awslint:props-default-doc:aws-cdk-lib.aws_ec2.LaunchTemplateProps.versionDescription] Optional property must have @default documentation 
aws-cdk-lib: Error: /codebuild/output/src1195990305/src/github.com/aws/aws-cdk/tools/@aws-cdk/cdk-build-tools/bin/cdk-awslint exited with error code 1
aws-cdk-lib: Build failed.!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

**Review by go-to-k — CHANGES_REQUESTED** [review]

File: packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts [hunk 2]
```diff
@@ -221,6 +221,15 @@ export interface LaunchTemplateProps {
    */
   readonly launchTemplateName?: string;
 
+  /**
+   * A description for the first version of the launch template.
+   * The version descrioption must be maximum 255 characters long, including hyphens (-),
+   * underscores (_), spaces, and tabs.
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts:227]: `including hyphens (-),`
`   * underscores (_), spaces, and tabs.`

Is there documentation available to confirm this? (I could not find that in the CFn documentation.)

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-launchtemplate.html#cfn-ec2-launchtemplate-versiondescription

File: packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts [hunk 3]
```diff
@@ -221,6 +221,15 @@ export interface LaunchTemplateProps {
    */
   readonly launchTemplateName?: string;
 
+  /**
+   * A description for the first version of the launch template.
+   * The version descrioption must be maximum 255 characters long, including hyphens (-),
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts:226]: A line break would make it clearer.

```suggestion
   * A description for the first version of the launch template.
   * 
   * The version descrioption must be maximum 255 characters long, including hyphens (-),
```

**Comment by yuppe-kyupeen** [comment]
@go-to-k 
Thank you for the review!
I have addressed the points you mentioned.I apologize for any oversights.
Please review the changes.Thank you!

File: packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts [hunk 4]
```diff
@@ -221,6 +221,17 @@ export interface LaunchTemplateProps {
...
   readonly launchTemplateName?: string;
 
+  /**
+   * A description for the first version of the launch template.
+   *
+   * The version descrioption must be maximum 255 characters long.
+   *
+   * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-launchtemplate.html#cfn-ec2-launchtemplate-versiondescription
+   *
+   * @default - The version description is not set if not specified.
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts:231]: I think it could be simpler.

```suggestion
   * @default - no description
```

**Review by go-to-k — CHANGES_REQUESTED** [review]

**Review by go-to-k — APPROVED** [review]
Thanks for the changes. Approved.

**Comment by yuppe-kyupeen** [comment]
@go-to-k 
thank you!

File: packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts [hunk 5]
```diff
@@ -221,6 +221,17 @@ export interface LaunchTemplateProps {
    */
   readonly launchTemplateName?: string;
 
+  /**
+   * A description for the first version of the launch template.
+   *
+   * The version descrioption must be maximum 255 characters long.
```
Comment (gracelu0) [inline packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts:227]:
```suggestion
   * The version description must be maximum 255 characters long.
```

**Review by gracelu0 — COMMENTED** [review]
Hi @yuppe-kyupeen , thanks for your contribution. Just one small typo to fix, otherwise LGTM!

**Comment by yuppe-kyupeen** [comment]
Hi @gracelu0, thanks for your review and suggestion! 👍

**Review by gracelu0 — APPROVED** [review]

**Pull request merged**